certbot/certbot/display/util.py

```diff
@@ -202,3 +206,36 @@ def assert_valid_call(prompt, default, cli_flag, force_interactive):
         msg += ("\nYou can set an answer to "
                 "this prompt with the {0} flag".format(cli_flag))
     assert default is not None or force_interactive, msg
+
+
+# This class takes a similar approach to the cryptography project to deprecate attributes
+# in public modules. See the _ModuleWithDeprecation class here:
+# https://github.com/pyca/cryptography/blob/91105952739442a74582d3e62b3d2111365b0dc7/src/cryptography/utils.py#L129
+class _DisplayUtilDeprecationModule:
+    """
+    Internal class delegating to a module, and displaying warnings when attributes
+    related to deprecated attributes in the certbot.display.util module.
+    """
+    def __init__(self, module):
+        self.__dict__['_module'] = module
+
+    def __getattr__(self, attr):
+        if attr in ('FileDisplay', 'NoninteractiveDisplay', 'SIDE_FRAME', 'input_with_timeout',
+                    'separate_list_input', 'summarize_domain_list', 'obj', 'WIDTH', 'HELP', 'ESC'):
```

Comment:
I don't think we need `obj` here as it's just an import and was never a documented part of the API.
```suggestion
                    'separate_list_input', 'summarize_domain_list', 'WIDTH', 'HELP', 'ESC'):
```
Other than that, this LGTM!